### PR TITLE
defcustom projectile-rails-{rails,spring}-command

### PR DESCRIPTION
**Pull request by esjee**
I would like this functionality for myself, so I tried to implement it. I'm very new to writing elisp, so any pointers would be much appreciated.

I'm struggling installing packages locally, so this is completely untested. I saw that there was a travis config, so I'm hoping that will be able to detect any issues.

### CODE REVIEW

**Comment by esjee** [comment]
This failure seems unrelated, https://travis-ci.org/asok/projectile-rails reports that the same test is failing on master.

**Comment by asok** [comment]
Cool you want to contribute.
But can I ask in what scenario you would want to run rails or spring in other way then via `bundle exec`?

**Comment by esjee** [comment]
Well, to be completely honest with you, I'm running into a problem that I don't understand. This happens if I run `:Rconsole`:

```
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    bundler (>= 1.10.2) ruby

  Current Bundler version:
    bundler (1.8.5)

This Gemfile requires a different version of Bundler.
Perhaps you need to update Bundler by running `gem install bundler`?

Process ruby exited abnormally with code 6
```

I'm having a hard time understanding where this bundler version is coming from. I'm guessing it has something to do with setting the path in `zshrc`, and emacs not sourcing that. Moreover, I'm also running into problems with another emacs plugin if I use spring outside of emacs (e.g. in a console).

For that other plugin, I solved the problem by using generated binstubs (`bundle exec spring binstub`). I'd like to do the same here.

**Comment by asok** [comment]
For that you might want to look at: https://github.com/asok/projectile-rails#running-ruby-gems
Basically you have to make Emacs use correct version of ruby for you. There are packages for integrating with rvm, rbenv and chruby. Normally you call `{rvm|rbenv|chruby}-use-corresponding` or `{rvm|rbenv|chruby}-use` and PATH that Emacs useses is set correctly. Also you might want to install exec-path-from-shell even if you are on Linux.

**Comment by esjee** [comment]
Thank you, I've managed to get rid of the bundler issue. I'm sorry to have asked something that I could have figured out for myself if I read the README more carefully. :)

However, I can still notice a significant difference between using binstubs (`bin/rails console`) versus bundle (`bundle exec rails console`). On my machine, the binstub is much faster to start than `bundle exec`.

I realize this is not a completely representative test, but here's the numbers I'm seeing:

```
bin/rails c  0,15s user 0,03s system 28% cpu 0,641 total
bundle exec rails c  2,71s user 0,21s system 88% cpu 3,303 total
```
The difference is roughly ~2 seconds. It's worth noting that before running with `time`, I ran both commands a couple of times.

@asok do you think there's still something I'm missing here, or could there be an improvement by using binstubs over bundle exec?

**Comment by asok** [comment]
> Thank you, I've managed to get rid of the bundler issue. I'm sorry to have asked something that I could have figured out for myself if I read the README more carefully. :)

No worries. It happens to me all the time :)

Well I guess under some circumstances someone would like to setup those commands to something else than `bundle exec`. Like when using binstubs as you mentioned. I'll accept it. Many thanks!

**Comment by asok** [comment]
Ah one small thing. Can you make the same for the `zeus` case?

**Comment by esjee** [comment]
Cool, thanks for your understanding. :)

I implemented the same for `zeus`, and also came across a couple of cases that I missed (such as a leading space after "destroy" -> "destroy ").

File: projectile-rails.el [hunk 1]
```diff
@@ -207,6 +207,21 @@
 (defcustom projectile-rails-discover-bind "s-r"
   "The :bind option that will be passed `discover-add-context-menu' if available")
 
+(defcustom projectile-rails-rails-command "bundle exec rails"
```
Comment (asok) [inline projectile-rails.el]: @esjee maybe name `projectile-rails-vanilla-command` would be better here?

Comment (esjee) [inline projectile-rails.el]: Thanks for the suggestion! I like that better than `rails-rails`, which reads a bit weird.

**Comment by esjee** [comment]
By the way, do you have any advice on how to do emacs development? Specifically, I have no clue how I can (easily) run local emacs extensions, common practices, anything like that. Do you perhaps have any pointers, or a link to some page?

Not that I didn't try googling around, but every resource I found was pretty daunting, and I can't imagine there isn't some easier way. :smile:

**Comment by asok** [comment]
> By the way, do you have any advice on how to do emacs development? Specifically, I have no clue how I can (easily) run local emacs extensions, common practices, anything like that. Do you perhaps have any pointers, or a link to some page?

Hard to point the best resources because there are so many of them. But I can think of some (in this particular order):
* http://steve-yegge.blogspot.com/2008/01/emergency-elisp.html
* https://www.masteringemacs.org/article/evaluating-elisp-emacs 
* http://emacsrocks.com/ if you scroll to the bottom you'll have the section "Extending Emacs"

Also the emergency elisp is very useful though quite outdated. Since the time it was written some nice libraries were created:
* https://github.com/magnars/dash.el
* https://github.com/magnars/s.el
* https://github.com/rejeep/f.el

Also reading code is quite educating. I remember learned a lot and picked a lot of ideas from [rspec-mode](https://github.com/pezra/rspec-mode/blob/master/rspec-mode.el) and [projectile](https://github.com/bbatsov/projectile/blob/master/projectile.el).

Also after you feel like home in Emacs you can grab a lot of new packages from http://planet.emacsen.org

**Pull request merged**